### PR TITLE
Aggregation for closed-loop composite metrics

### PR DESCRIPTION
**Pull request by perone**
Adding an aggregator for composite metrics on the closed-loop evaluation.

### CODE REVIEW

File: l5kit/l5kit/cle/composite_metrics.py [hunk 1]
```diff
@@ -130,3 +132,81 @@ def compute(self, metric_results: Dict[str, torch.Tensor],
...
+    """Aggregates composite metric results locally and across multiple
+    distributed nodes.
+
+    :param reduce_mode: the reduce operation to be applied both into
+                        local scenes and also across distributed
+                        nodes.
+    """
+
+    #: Mapping from ReduceMode to torch distributed ops
+    torch_reduce_mapping = {
```
Comment (lucabergamini) [inline l5kit/l5kit/cle/composite_metrics.py:172]: not used as far as I can tell

File: l5kit/l5kit/cle/composite_metrics.py [hunk 2]
```diff
@@ -130,3 +132,81 @@ def compute(self, metric_results: Dict[str, torch.Tensor],
...
+        ReduceMode.SUM: torch.distributed.ReduceOp.SUM
+    }
+
+    def __init__(self, reduce_mode: ReduceMode = ReduceMode.SUM):
+        # Supporting only ReduceMode.SUM for the moment
+        if reduce_mode != ReduceMode.SUM:
```
Comment (lucabergamini) [inline l5kit/l5kit/cle/composite_metrics.py:178]: shouldn't we check in `ReduceMode` ?

File: l5kit/l5kit/cle/composite_metrics.py [hunk 3]
```diff
@@ -130,3 +132,81 @@ def compute(self, metric_results: Dict[str, torch.Tensor],
...
+                         Dict[int, Dict[str, float]]) -> Dict[str, Any]:
+        """Aggregate the scenes locally on each node. This method will just
+        sum the number of invalid scenes across all scenes in the node.
+
+        :param scene_composite_metric_results: results from composite metric
+                                               outputs per scene
+        :returns: a dictionary with the composite metric name as keys
+                  and the reduction of the composite metrics
+        """
+        if self.reduce_mode == ReduceMode.SUM:
```
Comment (lucabergamini) [inline l5kit/l5kit/cle/composite_metrics.py:192]: this is already checked in init right?

**Review by lucabergamini — COMMENTED** [review]
minors

File: l5kit/l5kit/cle/composite_metrics.py (hunk 3 above)
Comment (perone) [reply l5kit/l5kit/cle/composite_metrics.py:186]: Yes, it is just there as a placeholder to add more reductions later. I will leave as is just to not diverge from the one in avjadoo.

File: l5kit/l5kit/cle/composite_metrics.py (hunk 2 above)
Comment (perone) [reply l5kit/l5kit/cle/composite_metrics.py:172]: Not sure if I understood, check where you mean ?

File: l5kit/l5kit/cle/composite_metrics.py (hunk 1 above)
Comment (perone) [reply l5kit/l5kit/cle/composite_metrics.py]: Fixed

File: l5kit/l5kit/cle/composite_metrics.py (hunk 2 above)
Comment (lucabergamini) [reply l5kit/l5kit/cle/composite_metrics.py:172]: Shouldn’t we check if `reduce_mode` is among those in `ReduceMode`?

**Review by lucabergamini — APPROVED** [review]
LGTM

File: l5kit/l5kit/cle/composite_metrics.py (hunk 2 above)
Comment (perone) [reply l5kit/l5kit/cle/composite_metrics.py:172]: Ah ok, makes sense, will fix it.

**Review by lucabergamini — APPROVED** [review]

**Pull request merged**